hledger-lib/Hledger/Read/Common.hs

```diff
@@ -763,8 +763,9 @@ rawnumberp = do
         leadingDigits <- some digitChar
         option (Nothing, [leadingDigits]) . try $ do
             firstSep <- oneOf sepChars <|> whitespaceChar
-            groups <- some digitChar `sepBy1` char firstSep
-            return (Just firstSep, leadingDigits : groups)
+            secondGroup <- some digitChar
+            otherGroups <- many $ try $ char firstSep *> some digitChar
```

Comment:
Hmm... Interesting how this is different from previous code that suppose to produce list of one or more elements with non-zero list of digit characters separated by `firstSep`? Am I missing something?

Reply:
Yes, the new code also parses a list of one or more numbers separated by `firstSep`, where each number is a non-zero list of digit characters. The difference is only that I have added `try` so that a separator and the following digits are parsed together as an atomic unit. The old code failed to parse the above examples because it interpreted trailing whitespace as a separator.  With the new code, whitespace will only be interpreted as a separator if digits follow it.

Reply:
Hmm... Yep. Confirmed with test-case

```
### Failure in /home/nikolay/contrib/hledger/hledger-lib/Hledger/Read/Common.hs:757: expression `parseTest rawnumberp "1 000 USD"'
expected: RawNumber (Just ' ') ["1","000"] Nothing
 but got: 1:2:
          unexpected space
          expecting digit
```

Apparently `sepBy` in `megaparsec` expects that if separator were parsed successfully then next entry is guaranteed to parse correctly.

Though I still cannot understand error message. `notFollowedBy $  (whitespaceChar >> digitChar)` should say something about unexpected space followed by digit.

Reply:
It seems that the problem is that the remaining text that the `notFollowedBy ...` line is given to parse is ` 000 USD`, which triggers its second alternative. The first section of `rawnumberp` doesn't consume the ` 000` due to `try` in `option (...) . try $ do` that encloses the failure on the `groups <- some digitChar ...` line.